app/Http/Controllers/V2/Terrafund/TerrafundEditGeometryController.php

```diff
@@ -34,23 +34,22 @@ public function updateProjectCentroid($polygonGeometry)
     {
         try {
             $sitePolygon = SitePolygon::where('poly_id', $polygonGeometry->uuid)->first();
+            if (!$sitePolygon) {
+              Log::warning("Site polygon with UUID $polygonGeometry->uuid not found.");
+              return null;
+            }
+            $relatedSite = Site::where('uuid', $sitePolygon->site_id)->first();
+            $project = Project::where('id', $relatedSite->project_id)->first();
```

Comment:
SitePolygon has a relationship to Project through Site, so this whole thing can be a single query using built-in helpers:
```
$project = $sitePolygon->project
```

Reply:
is changed to single query now.